### DISCUSSION TITLE
動的ローディング形式使用時のアプリケーション最大サイズについて

### DISCUSSION THREAD

**Discussion by okazakikoji** [toppers/users-forum#80] [ET-Robocon]
プログラムをEV3にロードした際、「Application is too large.」とエラー表示されました。
以下のページに「ローダの設定に依存しますが、デフォルトは1MB」を記載があるのに対し現状1.5MBと超えていますのでこれが原因かと思われます。
https://dev.toppers.jp/trac_user/ev3pf/wiki/SampleProgram

スタンドアローン形式でこの点回避できる旨後述されていますが、一方、動的ローディング形式に対しても「ローダの設定に依存しますが」という記載があります。ここは何か設定をすることで1MB超えたものを動かすことができるのでしょうか。
もし対応方法ありましたらご教授いただけると幸いです。
![image](https://user-images.githubusercontent.com/63992950/178613964-a253af3b-83b9-469a-acee-c6d7f56e5673.png)

**Comment by YixiaoLi** [comment]
動的ローディングのリソース上限は、[target/ev3_gcc/ev3.h](https://github.com/ev3rt-git/ev3rt-hrp3/blob/master/target/ev3_gcc/ev3.h#L79-L91)の「Loadable application module configuration (Dynamic loading)」に定義されています。
TMAX_APP_BINARY_SIZEをデフォルトの(1024 * 1024)から(2048 * 1024)に修正すれば、2MBのバイナリをロードできるようになります。
なお、ev3.hの内容を修正した後、下記コマンドでローダのuImageをビルドして使う必要があります。
```
# Build loader uImage
cd ev3rt-hrp3/sdk/firmware
make img=loader
```

**Reply by okazakikoji** [reply]
ご回答いただきありがとうございます。画像のようなエラーは出なくなりました。
※しかし、サイズが大きいアプリは読み込み後動かないので、問題切り分けてまた質問させていただくかもしれません。

**Reply by okazakikoji** [reply]
無事に1MB超のアプリを実行することができておりますのでご連絡しておきます。ありがとうございました。
